File: Code Samples/OOP/OOP I - Nikita Choudhari/Java/Person.java
// Inheritance
class Animal {
    void eat() {
        System.out.println("Animal is eating.");
    }
}

class Dog extends Animal {
    void bark() {
        System.out.println("Dog is barking.");
    }

    public static void main(String[] args) {
        Dog dog = new Dog();
        dog.eat();  // Inherited from Animal
        dog.bark(); // Defined in Dog
    }
}

// Encapsulation
class Person {
    private String name;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public static void main(String[] args) {
        Person person = new Person();
        person.setName("John");
        System.out.println("Person's name is " + person.getName());
    }
}

File: Code Samples/OOP/OOP I - Nikita Choudhari/Python/Person.py
# Inheritance
class Animal:
    def eat(self):
        print("Animal is eating.")

class Dog(Animal):
    def bark(self):
        print("Dog is barking.")

dog = Dog()
dog.eat()  # Inherited from Animal
dog.bark() # Defined in Dog

# Encapsulation
class Person:
    def __init__(self):
        self.__name = ""

    def get_name(self):
        return self.__name

    def set_name(self, name):
        self.__name = name

person = Person()
person.set_name("John")
print("Person's name is", person.get_name())
